src/cmdstan/arguments/arg_log_prob.hpp

```diff
@@ -0,0 +1,26 @@
+#ifndef CMDSTAN_ARGUMENTS_ARG_LOG_PROB_HPP
+#define CMDSTAN_ARGUMENTS_ARG_LOG_PROB_HPP
+
+#include <cmdstan/arguments/arg_log_prob_unconstrained_params.hpp>
+#include <cmdstan/arguments/arg_log_prob_constrained_params.hpp>
+#include <cmdstan/arguments/arg_log_prob_jacobian_adjust.hpp>
+#include <cmdstan/arguments/categorical_argument.hpp>
+
+namespace cmdstan {
+
+class arg_log_prob : public categorical_argument {
+ public:
+  arg_log_prob() {
+    _name = "log_prob";
+    _description
+        = "Return the log-probability and its gradients, given supplied "
```

Comment:
log-probability -> log density up to a constant

P.S.  The term `log_prob` was an unfortunate choice based on BDA's use of "probability function" to refer to either a probability density function or probability mass function.  It's technically a mixed probability function because some of the variables can be discrete, but we use "density" as the general term for that (in retrospect, I think "_lpdf" and "_lpmf" distinctions were a mistake because with tuples we'll be able to have probability functions that are mixed.